beacon_node/network/src/subnet_service/mod.rs

```diff
@@ -132,20 +133,21 @@ impl<T: BeaconChainTypes> SubnetService<T> {
 
         // Build the list of known permanent subscriptions, so that we know not to subscribe or
         // discover them.
-        let mut permanent_attestation_subscriptions = HashSet::default();
+        let mut permanent_attestation_subscriptions = BitVector::new();
         if config.subscribe_all_subnets {
             // We are subscribed to all subnets, set all the bits to true.
             for index in 0..beacon_chain.spec.attestation_subnet_count {
-                permanent_attestation_subscriptions
-                    .insert(Subnet::Attestation(SubnetId::from(index)));
+                permanent_attestation_subscriptions.set(index as usize, true)
+                    .expect("Subnet index should be within bounds");
```

Comment:
Even though this _should_ never fail, we want to avoid using `unwrap` or `expect` like this so we will need to refactor this to include proper error handling